<a href="https://colab.research.google.com/github/traviolus/DataSci/blob/master/Activity1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


In [4]:
cat_id = pd.read_csv('/content/drive/My Drive/Colab Notebooks/cat_id.csv')
cat_id.head()

2 Japanese
0  3     Thai
1  4  Chinese
2  5  Italian
3  6   Korean
4  7  Vietnam

In [5]:
wongnai = pd.read_csv('/content/drive/My Drive/Colab Notebooks/wongnai.csv')
wongnai.head()

Unnamed: 0  id  ... number_of_checkins  all_time_score
0           0   2  ...                 32    22487.872358
1           1   3  ...                  0       92.900000
2           2   4  ...                173   239274.950390
3           3   5  ...                 21    13307.512755
4           4   6  ...                 14    19606.094680

[5 rows x 15 columns]

In [34]:
wongnai.isnull().sum()

Unnamed: 0                           0
id                                   0
name                                 1
category_id                         60
city_id                            206
latitude                             0
longitude                            0
price_range                     100196
avg_price                       278496
number_of_bookmarks                  0
number_of_favorite_bookmarks         0
top                                  0
best_of_wongnai                 283128
number_of_checkins                   0
all_time_score                       0
dtype: int64

In [32]:
wongnai2 = wongnai.dropna()
wongnai2.shape

(264, 15)

# 1. ข้อมูลนี้มีปัญหาในเรื่องข้อมูลไม่ครบถ้วน หรือติดลบ ที่ไหนบ้าง และเมื่อตัดแถวที่มีข้อมูล NaN หรือ ติดลบแล้วเหลือเท่าไหร่
ข้อมูลที่มีปัญหาในแต่ละคอลัมน์ มีจำนวนตามที่แสดงจากโค้ด wongnai.isnull().sum()

และเมื่อตัดแถวที่มีปัญหาแล้ว เหลือ 264 แถว

In [35]:
wongnai.sort_values(by='best_of_wongnai').head(5)['name']

141447    Water Library Brasserie
135966          Caturday cat cafe
129258                   Chef Man
141406               Din Tai Fung
118873                 Eat Am Are
Name: name, dtype: object

# 2. จงแสดงชื่อร้านอาหาร 5 อันดับแรกในกลุ่ม Best of Wongnai
1. Water Library Brasserie
2. Caturday Cat Cafe
3. Chef Man
4. Din Tai Fung
5. Eat Am Are

In [23]:
groups = wongnai.reset_index().groupby("category_id")
groups.size()

category_id
2.0        5713
3.0       39458
4.0        6860
5.0        1873
6.0        1459
          ...  
1201.0       10
1202.0        8
1300.0      269
1400.0       34
1500.0      162
Length: 129, dtype: int64

In [29]:
wongnai.groupby('category_id').size().sort_values(ascending=False).head(3)

category_id
3.0     39458
14.0    29814
30.0    29759
dtype: int64

# 4. จงระบุกลุ่มร้านอาหารทีมีผู้ใช้ไปใช้บริการมากที่สุด 3 อันดับแรก
1. 3.0 
2. 14.0
3. 30.0
